Update OIDC authentication documentation (#336)
rancher/stackstate-product-docs: docs/latest/modules/en/pages/setup/security/authentication/oidc.adoc

----

> In HA deployments, the api component handles configuration differently. Use the following:

Reviewer: We can drop this sentence (it seems to be left-over from a rewrite and doesn't "end")

=== Other OIDC providers

> To configure SUSE Observability to use an OIDC authentication provider on Kubernetes, you need to add OIDC details and user role mapping to the file `authentication.yaml`. For example:

Reviewer: ```suggestion
To configure SUSE Observability to use any other OIDC provider, you need to add OIDC details and user role mapping to the file `authentication.yaml`. For example:
```

----

> In HA deployments, the api component handles configuration differently. Use the following:

Author: ```suggestion
```
